praw/models/reddit/draft.py

```diff
@@ -0,0 +1,331 @@
+        cls,
+        flair_id: Optional[str] = None,
+        flair_text: Optional[str] = None,
+        is_public_link: Optional[bool] = None,
+        nsfw: Optional[bool] = None,
+        original_content: Optional[bool] = None,
+        selftext: Optional[str] = None,
+        send_replies: Optional[bool] = None,
+        spoiler: Optional[bool] = None,
+        subreddit: Optional[
+            Union["praw.models.Subreddit", "praw.models.UserSubreddit"]
+        ] = None,
+        title: Optional[str] = None,
+        url: Optional[str] = None,
+        **draft_kwargs,
+    ):
+        data = {
+            "body": selftext or url,
+            "flair_id": flair_id,
+            "flair_text": flair_text,
+            "is_public_link": is_public_link,
+            "kind": "markdown" if selftext is not None else "link",
+            "nsfw": nsfw,
+            "original_content": original_content,
+            "send_replies": send_replies,
+            "spoiler": spoiler,
+            "title": title,
+        }
+        if subreddit:
+            data.update(
+                {
+                    "subreddit": subreddit.fullname,
+                    "target": "profile"
+                    if subreddit.display_name.startswith("u_")
+                    else "subreddit",
+                }
+            )
+        data.update(draft_kwargs)
+        data = {key: value for key, value in data.items() if value is not None}
+        return data
+
+    def __init__(
+        self,
+        reddit: "praw.Reddit",
+        id: Optional[str] = None,  # pylint: disable=redefined-builtin
+        _data: Dict[str, Any] = None,
+    ):
+        """Construct an instance of the Draft object."""
+        if (id, _data).count(None) != 1:
+            raise TypeError("Exactly one of `id` or `_data` must be provided.")
+        fetched = False
+        if id:
+            self.id = id
+        else:
+            if len(_data) > 1:
+                if _data["kind"] == "markdown":
+                    _data["selftext"] = _data.pop("body")
+                elif _data["kind"] == "link":
+                    _data["url"] = _data.pop("body")
+                fetched = True
+        super().__init__(reddit, _data=_data, _fetched=fetched)
+
+    def __eq__(self, other: "Draft"):
+        """Return whether the other instance equals the current."""
+        if isinstance(other, str):
+            return other.lower() == str(self).lower()
+        body = True
+        if self._fetched and other._fetched:
+            if self.kind == other.kind == "markdown":
+                body = self.selftext == other.selftext
+            elif self.kind == other.kind == "link":
+                body = self.url == other.url
+        return all(
+            [
+                isinstance(other, self.__class__),
+                self.id.lower() == other.id.lower(),
+                body,
+            ]
+        )
+
+    def __hash__(self):  # not sure why this doesn't inherit correctly
```

Comment:
I'm curious about this happening too. Something else must be an issue.

Reply:
I have no idea and I'm out of ideas why it doesn't inherit correctly.

Reply:
What is returned when this method isn't overwritten?

Reply:
`Draft is not hashable`